src/printer.rs

```diff
@@ -86,16 +85,20 @@ impl Printer {
 
         let mut header_string = String::new();
         for (key, value) in headers {
-            let key = key.to_string();
-            let value = value.to_str().unwrap();
-            writeln!(&mut header_string, "{}: {}", key, value).unwrap();
+            header_string.push_str(key.as_str());
+            header_string.push_str(": ");
+            match value.to_str() {
+                Ok(value) => header_string.push_str(value),
+                Err(_) => header_string.push_str(&format!("{:?}", value)),
+            }
```

Comment:
Rather than being silent, I think it is ok to panic if an invalid header value is detected
```
pub fn to_str(&self) -> Result<&str, ToStrError>
------------------------------------------------
Yields a &str slice if the HeaderValue only contains visible ASCII chars.
```

Reply:
Hmm. Such a value is weird but technically valid:

> In practice, HTTP header field values are usually valid ASCII. However, the HTTP spec allows for a header value to contain opaque bytes as well. In this case, the header field value is not able to be represented as a string.

I figure that if reqwest allowed it to be constructed it's best to just print it. It's probably only relevant for responses though.

And making the program more verbose should probably never cause it to fail, it'd be bad for it to crash when you tried to see more. But I didn't think of that while writing so I'm not sure that's upheld.

Reply:
On second thought, it's upheld in functions that return an `io::Result`, because those can only fail for I/O reasons.

`print_json()` is the only one that can fail for non-I/O reasons, if `indent_json()` fails, because the JSON is invalid. But I think the user would want to know what's invalid about it, and rejecting it based on configuration is bad. Maybe it should just fall back to the unindented original? With or without a warning? What do you think?

Reply:
>And making the program more verbose should probably never cause it to fail, it'd be bad for it to crash when you tried to see more.

👍

>Maybe it should just fall back to the unindented original? With or without a warning? What do you think?

Sounds good. Since `print_json()` will not write anything until `indent_json()` succeeds, I think the simplest option would be to do the fallback inside `print_json()` i.e `indent_json(text).unwrap_or(text)`.